### ISSUE TITLE
Self-hosted data layer (and rich-text resolution) breaks on latest version of Tina

### ISSUE THREAD

**Issue by coreyaus** [tinacms/tinacms#3771]
### Describe the issue that you're seeing. Any Loom videos or screenshots usually help a lot!

We experienced a couple of issues when updating to the latest versions of all the Tina packages. We were able to resolve the first by patching Tina, but I could only work around the second without actually resolving it:

1. The new `configManager` runs a couple of checks that fail on self-hosted versions of Tina, specifically:
    - `await this.checkClientInfo(configManager, apiURL)` on this line: https://github.com/tinacms/tinacms/blob/main/packages/%40tinacms/cli/src/next/commands/build-command/index.ts#L118
    - `await this.checkGraphqlSchema(configManager, database, apiURL)` on this line: https://github.com/tinacms/tinacms/blob/main/packages/%40tinacms/cli/src/next/commands/build-command/index.ts#L120
    - I'm not sure whether the intent would be to modify those checks to also cater for self-hosted environments, but for now my approach has been to wrap those 2 lines in the following conditional: `if(!configManager.hasSelfHostedConfig()){ ... }`

2. The resolution of `rich-text` fields fails with self-hosting
    - The screenshot below shows the error messages that arise due to this issue
    - It seems for some reason the `parseMDX` function on the following lines is not available in all environments: https://github.com/tinacms/tinacms/blob/main/packages/%40tinacms/graphql/src/resolver/index.ts#L771-L777
    - ~I haven't been able to work out how to resolve or patch this issue. For now I've just changed the `rich-text` fields in my schema to `string` fields to successfully run the latest version of Tina. However that's obviously not a proper solution as it removes all the formatting from that rich-text content~
    - ~Are there any immediate ideas you can suggest when viewing those error messages below? It seems something about the recent updates mean the `parseMDX` import from the `@tinacms/mdx` package no longer works in all cases - I'm not sure if that suggests a missing peerDependency, or something askew in the build/bundling process - let me know if you have any thoughts and I'm happy to help out if you'd like me to test your suggestions~
    - **UPDATE / SOLUTION:** I was able to resolve this issue by adding the following setting to my `next.config.js`:
        ```js
        experimental: { serverComponentsExternalPackages: ['@tinacms/mdx'] }
        ```
        - The full notes on that feature can be seen here: https://beta.nextjs.org/docs/api-reference/next-config#servercomponentsexternalpackages. In short, it seems the `@tinacms/mdx` package uses Node.js specific features, so it's necessary to opt that dependency out of Server Components bundling so it uses the native Node.js `require`

![error-loading-parseMDX-function](https://user-images.githubusercontent.com/7799612/229518484-5113be36-60ea-4d52-b5ba-be591629f083.png)

### Other notes

- I can successfully run the `tinacms audit` command, so there is nothing wrong with the content or the schema (and the `parseMDX` function is obviously imported successfully in the environment where the audit runs)
- When first running `yarn dev` after upgrading Tina we hit the following error: "Transforming `const` to the configured target environment ("es5") is not supported yet". We resolved that by updating our `tsconfig.json` from `"target": "es5"` to `"target": "es6"` (noting here in case it has any bearing on the two points above)

<img width="673" alt="image" src="https://user-images.githubusercontent.com/7799612/229540826-f72a28d2-7570-436f-9878-f894e229458a.png">

### Reproduction

https://github.com/tinacms/tina-self-hosted-demo/pull/40

### Steps to reproduce

Ultimately the steps to reproduce have just been to:

- upgrade to the latest versions of Tina on a repository that uses self-hosting
- ensure that the schema includes a `rich-text` field, and the page being loaded includes content for at least 1 rich-text field

I'm posting this issue now in case there's anything that's obvious to the core Tina team that could have caused this issue as part of the recent changes. It's the end of my day here but if nothing comes to mind I can of course create a fresh Tina repository with self-hosting support and upgrade it to the latest version of Tina to demonstrate what I'm seeing in a public repo 😄 

### System Info

```shell
- System:
    - OS: macOS 13.2.1
    - CPU: (4) x64 Intel(R) Core(TM) i7-7660U CPU @ 2.50GHz
    - Memory: 521.84 MB / 16.00 GB
    - Shell: 5.8.1 - /bin/zsh
- Binaries:
    - Node: 16.14.2 - ~/.nvm/versions/node/v16.14.2/bin/node
    - Yarn: 3.2.1 - ~/GitHub/sitejoy/node_modules/.bin/yarn
    - npm: 8.5.0 - ~/.nvm/versions/node/v16.14.2/bin/npm
- Browsers:
    - Chrome: 111.0.5563.146
    - Firefox: 110.0.1
    - Safari: 16.3
- npmPackages:
    - @tinacms/auth: 1.0.3 => 1.0.3
    - @tinacms/cli: 1.5.3 => 1.5.3
    - @tinacms/datalayer: 1.2.4 => 1.2.4
    - tinacms: 1.4.3 => 1.4.3
```

### Validations

- [X] Follow our [Code of Conduct](https://github.com/tinacms/tinacms/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [docs](https://tina.io/docs).
- [X] Check that there isn't [already an issue](https://github.com/tinacms/tinacms/issues) that reports the same bug.

**Comment by logan-anderson** [comment]
@coreyaus  I am able to re-produce the issue and am looking into it. Thanks for reporting @coreyaus.

**Comment by coreyaus** [comment]
Massive thanks mate - you were incredibly quick to respond!

I've updated my issue above as I was able to resolve the second point too by adding the following setting to my `next.config.js`:

```js
experimental: { serverComponentsExternalPackages: ['@tinacms/mdx'] }
```

The full notes on that feature can be seen here: https://beta.nextjs.org/docs/api-reference/next-config#servercomponentsexternalpackages. In short, it seems the `@tinacms/mdx` package uses Node.js specific features, so it's necessary to opt that dependency out of Server Components bundling so it uses the native Node.js `require`

I assume that issue probably only impacted projects using the self-hosting and also running Next.js 13+ (which introduced the new `app` directory)

On the first point, if you're happy with the approach of just skipping the `checkClientInfo` and `checkGraphqlSchema` functions when self-hosting is enabled then I could open a PR to do that. Anything beyond that and I probably don't yet have the context on the Tina internals to update those functions to work with self-hosting sorry. Let me know if wrapping those in a conditional is all you think is needed at this stage and I can create a PR to do that (using the simple conditional below) 👍 

```js
if(!configManager.hasSelfHostedConfig()){ ... }
```

Happy taco Tuesday!
